src/libraries/System.Collections.Immutable/src/System/Collections/Frozen/String/KeyAnalyzer.cs

```diff
@@ -263,25 +245,34 @@ public AnalysisResults(bool ignoreCase, bool allAsciiIfIgnoreCase, int hashIndex
             public bool RightJustifiedSubstring => HashIndex < 0;
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static ReadOnlySpan<char> Slicer(this string s, int index, int count) => s.AsSpan((index >= 0 ? index : s.Length + index), count);
+
         private abstract class SubstringComparer : IEqualityComparer<string>
         {
-            public int Index;
-            public int Count;
-            public bool IsLeft;
+            public int Index;   // offset from left side (if positive) or right side (if negative) of the string
```

Comment:
Documenting how this (when positive) is the offset from front or (when negative) is the offset from the end of each string